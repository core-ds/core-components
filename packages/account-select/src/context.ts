import { createContext, useContext } from 'react';

export const AccountSelectContext = createContext<{
    setError: (error: string | null) => void;
}>({
    setError: () => {},
});

export const useAccountSelectContext = () => useContext(AccountSelectContext);
