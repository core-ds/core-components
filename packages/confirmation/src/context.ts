import { createContext } from 'react';

import { type TConfirmationContext } from './types';
import { ONE_DAY, ONE_MINUTE } from './utils';

const mockFn = () => undefined;

export const ConfirmationContext = createContext<TConfirmationContext>({
    alignContent: 'left',
    titleTag: 'h3',
    texts: {},
    state: 'INITIAL',
    screen: 'INITIAL',
    requiredCharAmount: 5,
    countdownDuration: ONE_MINUTE,
    timeLeft: ONE_MINUTE,
    tempBlockDuration: ONE_DAY,
    phone: '',
    hideCountdownSection: false,
    breakpoint: 1024,
    client: 'desktop',
    initialScreenHintSlot: null,
    onTempBlockFinished: mockFn,
    onInputFinished: mockFn,
    onChangeState: mockFn,
    onSmsRetryClick: mockFn,
    onChangeScreen: mockFn,
    onFatalErrorOkButtonClick: mockFn,
});
