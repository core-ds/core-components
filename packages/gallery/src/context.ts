import { createContext } from 'react';
import SwiperCore from 'swiper';

import { GalleryImage, ImageMeta } from './types';

export type GalleryContext = {
    view: 'desktop' | 'mobile';
    singleSlide: boolean;
    currentSlideIndex: number;
    images: GalleryImage[];
    imagesMeta: ImageMeta[];
    fullScreen: boolean;
    mutedVideo: boolean;
    playingVideo: boolean;
    hideNavigation: boolean;
    setHideNavigation: (hideNavigation: boolean) => void;
    initialSlide: number;
    setFullScreen: (fullScreen: boolean) => void;
    setMutedVideo: (muteVideo: boolean) => void;
    setPlayingVideo: (playingVideo: boolean) => void;
    setImageMeta: (meta: ImageMeta, index: number) => void;
    slideTo: (index: number) => void;
    slideNext: () => void;
    slidePrev: () => void;
    getSwiper: () => SwiperCore | undefined;
    setSwiper: (swiper: SwiperCore) => void;
    onClose: () => void;
    setCurrentSlideIndex?: (index: number) => void;
    getCurrentImage: () => GalleryImage | undefined;
    getCurrentImageMeta: () => ImageMeta | undefined;
};

const mockFn = () => undefined;

// eslint-disable-next-line @typescript-eslint/no-redeclare
export const GalleryContext = createContext<GalleryContext>({
    view: 'desktop',
    singleSlide: false,
    currentSlideIndex: 0,
    images: [],
    imagesMeta: [],
    fullScreen: false,
    mutedVideo: false,
    playingVideo: false,
    hideNavigation: false,
    setHideNavigation: mockFn,
    setMutedVideo: mockFn,
    setPlayingVideo: mockFn,
    initialSlide: 0,
    setFullScreen: mockFn,
    setImageMeta: mockFn,
    slideTo: mockFn,
    slideNext: mockFn,
    slidePrev: mockFn,
    getSwiper: mockFn,
    setSwiper: mockFn,
    onClose: mockFn,
    setCurrentSlideIndex: mockFn,
    getCurrentImage: mockFn,
    getCurrentImageMeta: mockFn,
});
