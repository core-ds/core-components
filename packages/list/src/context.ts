import { createContext, type ReactNode } from 'react';

import { type ColorMarkerType } from './types';

export type TListContext = {
    /**
     * Упорядоченный список
     */
    orderedList?: boolean;

    /**
     * Маркер
     * @default '–' for ul and 'decimal' for ol
     */
    markerType?: 'lower-alpha' | 'decimal' | string | ReactNode;

    /**
     * Цвет маркера
     */
    colorMarker?: ColorMarkerType;

    /**
     * Список обратного счета
     */
    reversed?: boolean;

    /**
     * Номер пункта
     */
    index?: number;

    /**
     * Начало отсчета элементов списка
     */
    start?: number;
};

export const ListContext = createContext<TListContext>({});
