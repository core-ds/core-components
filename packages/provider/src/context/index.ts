import { createContext } from 'react';

// Добавить типы из компонентов нет возможности, так как будет зацикленный импорт зависимостей
// eslint-disable-next-line @typescript-eslint/no-explicit-any
export type ComponentsContextValue = Record<string, any>;

export type CoreContextValue = {
    components?: ComponentsContextValue;
};

export const CoreComponentsContext = createContext<CoreContextValue>({});

export const CoreComponentsProvider = CoreComponentsContext.Provider;
