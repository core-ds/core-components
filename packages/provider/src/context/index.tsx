import React, { createContext, ReactNode } from 'react';

// Добавить типы из компонентов нет возможности, так как будет зацикленный импорт зависимостей
// eslint-disable-next-line @typescript-eslint/no-explicit-any
export type ComponentsContextValue = Record<string, any>;

export type CoreContextValue = {
    components?: ComponentsContextValue;
};

export const CoreComponentsContext = createContext<CoreContextValue>({});

type ProviderProps = {
    /**
     * Дочерние элементы
     */
    children?: ReactNode;
    /**
     * Конфиг ui
     */
    config: CoreContextValue;
};

export const CoreComponentsProvider = ({ children, config }: ProviderProps) => (
    <CoreComponentsContext.Provider value={config}>{children}</CoreComponentsContext.Provider>
);
