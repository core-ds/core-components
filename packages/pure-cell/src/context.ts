import { createContext } from 'react';

export type PureCellContextValue = {
    /** Направление */
    direction?: 'horizontal' | 'vertical';
    dataTestId?: string;
    setMainHover?: () => void;
    unsetMainHover?: () => void;
};

// eslint-disable-next-line @typescript-eslint/no-redeclare
export const PureCellContext = createContext<PureCellContextValue>({});
