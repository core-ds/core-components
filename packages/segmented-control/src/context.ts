import { createContext } from 'react';

import { IDType } from './typing';

export type ContextType = {
    onChange: (id: IDType) => void;
    colors?: 'default' | 'inverted';
};

export const SegmentedControlContext = createContext<ContextType>({
    onChange: () => null,
    colors: 'default',
});
