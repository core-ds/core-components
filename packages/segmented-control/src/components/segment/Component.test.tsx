import React from 'react';

import { fireEvent, render } from '@testing-library/react';
import { AScoresCircleMIcon } from '@alfalab/icons-glyph/AScoresCircleMIcon';

import { Segment, SegmentProps } from '../../index';
import { SegmentedControlContext } from '../../context';

const renderComponent = ({ id, title, ...restProps }: SegmentProps) => (
    <Segment id={id} title={title} {...restProps} />
);

describe('segment', () => {
    it('should display correctly', () => {
        const { container } = render(renderComponent({ id: 1, title: 'Title' }));

        expect(container).toMatchSnapshot();
    });

    it('should set className', () => {
        const className = 'customClassName';
        const { container } = render(renderComponent({ id: 1, title: 'Label', className }));

        expect(container.firstElementChild).toHaveClass(className);
    });

    it('should set dataTestId', () => {
        const dataTestId = 'custom-data-test-id';
        const { getByTestId } = render(renderComponent({ id: 1, title: 'Label', dataTestId }));

        expect(getByTestId(dataTestId)).toBeInTheDocument();
    });

    it('should call onChange on click', () => {
        const onChange = jest.fn();
        const dataTestId = 'dataTestIDD';

        const { getByTestId } = render(
            <SegmentedControlContext.Provider value={{ onChange }}>
                {renderComponent({ id: 1, title: 'Label', dataTestId })}
            </SegmentedControlContext.Provider>,
        );

        fireEvent.click(getByTestId(dataTestId));

        expect(onChange).toBeCalledTimes(1);
    });

    it('render with icon as title', () => {
        const { container } = render(
            renderComponent({ id: 1, title: React.createElement(AScoresCircleMIcon) }),
        );

        expect(container).toMatchSnapshot();
    });
});
