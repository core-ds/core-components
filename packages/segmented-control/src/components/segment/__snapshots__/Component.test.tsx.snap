// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`segment render with icon as title 1`] = `
<div>
  <button
    class="segment segment"
    type="button"
  >
    <svg
      fill="currentColor"
      focusable="false"
      height="24"
      role="img"
      viewBox="0 0 24 24"
      width="24"
    >
      <path
        d="M10.519 12.888h2.908L12.04 8.443h-.055l-1.467 4.445z"
      />
      <path
        clip-rule="evenodd"
        d="M22 12c0 5.523-4.477 10-10 10S2 17.523 2 12 6.477 2 12 2s10 4.477 10 10zm-8.085-4.377L17 17h-2.283l-.693-2.23h-4.13L9.147 17H7l3.237-9.377c.314-.91.68-1.623 1.858-1.623s1.522.716 1.82 1.623z"
        fill-rule="evenodd"
      />
    </svg>
  </button>
</div>
`;

exports[`segment should display correctly 1`] = `
<div>
  <button
    class="segment segment"
    type="button"
  >
    Title
  </button>
</div>
`;
